include/envoy/stream_info/stream_info.h

```diff
@@ -520,6 +520,16 @@ class StreamInfo {
    * @return request headers.
    */
   virtual const Http::RequestHeaderMap* getRequestHeaders() const PURE;
+
+  /**
+   * @param Upstream connection's ClusterName.
+   */
+  virtual void setUpstreamClusterName(const absl::string_view upstream_cluster_name) PURE;
```

Comment:
I think this would be a lot more useful if we snap the ClusterInfo shared_ptr. Is there any reason not to do that?

Reply:
Thanks changed to ClusterInfoConstSharedPtr...